Code/Framework/AzCore/AzCore/base.h

```diff
@@ -15,9 +15,19 @@
 
 #ifndef AZ_ARRAY_SIZE
```

Comment:
Should we make this universal? 
```
#ifdef AZ_ARRAY_SIZE
#undef AZ_ARRAY_SIZE
#endif
```

Reply:
Is there any place where `AZ_ARRAY_SIZE` is already defined? I don't see any. Maybe we should just remove the `#ifndef` check?

Reply:
Maybe there is some developer or team that wants to define AZ_ARRAY_SIZE to their own macro... I don't have a good reason why we need the `#ifdef`.
We can remove it

Reply:
Yeah, maybe Astra Zeneca